### PR TITLE
equation Objects

### PR DESCRIPTION
**Pull request by yairvogel**

### CODE REVIEW

File: ComplexEquation.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace OnBoardAssignment
+{
+    class ComplexEquation : Equation
```
Comment (shaybenh7) [inline ComplexEquation.cs:7]: public

File: SimpleEquation.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        {
+            Console.WriteLine("simple Equation");
+            Num1 = num1;
+            Num2 = num2;
+            Op = op;
+            Result = CalculateResult();
+        }
+
+        public SimpleEquation(int num1, int num2, char op)
+        : this(num1, num2, CharToOp(op)) { }
```
Comment (shaybenh7) [inline SimpleEquation.cs:27]: tab

File: SimpleEquation.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+namespace OnBoardAssignment
+{
+    class SimpleEquation : Equation
+    {
+        public int Num1 { get; set; }
+        public int Num2 { get; set; }
+
+        public int Result { get; set; }
+        public Operator Op { get; set; }
+        public const EquationType Type = EquationType.Simple;
```
Comment (shaybenh7) [inline SimpleEquation.cs:15]: where is this being used?

Comment (shaybenh7) [inline SimpleEquation.cs:15]: const comes first

File: RandomEquations.cs [hunk 4]
```diff
@@ -8,7 +10,34 @@ internal static class RandomEquations
...
+                _ => ' '
+            };
+
+            return opString;
+        }
+
+        private static Op CharToOp(char ch)
+        {
+            Op op = ch switch
+            {
```
Comment (shaybenh7) [inline RandomEquations.cs]: why are these two method being define here and in SimpleEquation.
please extract to a new static class called Utils and use it from there to avoid copy-paste

File: RandomEquations.cs [hunk 5]
```diff
@@ -8,7 +10,34 @@ internal static class RandomEquations
     {
         private static readonly Random Rand = new Random();
 
-        private static readonly char[] op = { '+', '-', '*', '/' };
+        private enum Op { Add, Sub, Mul, Div }
```
Comment (shaybenh7) [inline RandomEquations.cs]: why is this needed? why can't you use the enum you defined as Operator?

Comment (shaybenh7) [reply RandomEquations.cs]: if not needed, please extract to Utils class and put the enum there as public

File: SimpleEquation.cs (hunk 3 above)
Comment (yairvogel) [reply SimpleEquation.cs:15]: are you sure? The docs say otherwise. There's the option of the declaring `const EquationType..` without an access modifier, is it correct?

File: RandomEquations.cs (hunk 5 above)
Comment (yairvogel) [reply RandomEquations.cs]: I Deleted RandomEquations and moved its fuctionalities to the Equation Class.

Comment (yairvogel) [reply RandomEquations.cs]: There are no duplications now.

File: ComplexEquation.cs (hunk 1 above)
Comment (yairvogel) [reply ComplexEquation.cs:7]: done

File: SimpleEquation.cs (hunk 2 above)
Comment (yairvogel) [reply SimpleEquation.cs:27]: done

File: SimpleEquation.cs (hunk 3 above)
Comment (yairvogel) [reply SimpleEquation.cs:13]: My idea was to use it for the second step of the program, but maybe it's not the best practice.
I tried to define the Result getter to perform the calculation lazily, but the getter is readonly, is there a way to overpass it?

File: RandomEquations.cs (hunk 4 above)
Comment (yairvogel) [reply RandomEquations.cs]: fixed. I Deleted RandomEquations and moved its fuctionalities to the Equation Class.

File: SimpleEquation.cs (hunk 3 above)
Comment (shaybenh7) [reply SimpleEquation.cs:13]: I mean, const property comes first in the order of properties

Comment (shaybenh7) [reply SimpleEquation.cs:13]: public const EquationType Type = EquationType.Simple;

public int Num1 { get; set; }
public int Num2 { get; set; }
....
